### PR TITLE
test: refactor v2 sample test, move test samples to samples/test folder

### PR DESCRIPTION
**Pull request by Bobgy**
**Description of your changes:**
Reduce v2 sample test code duplication

Changes:
* Move test samples from `v2/test/samples` to `samples/test` folder
* Extract a test util that makes sample_test.py much more focused on the testing the pipeline

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?

    [Learn more about cherry-picking updates into the release branch](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#cherry-picking-pull-requests-to-release-branch).

### CODE REVIEW

File: v2/test/samples/util.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        run_response = run_result.wait_for_run_completion(10 * MINUTE)
+        run = run_response.run
+        from pprint import pprint
+        pprint(run_response.run)
+        print("Run details page URL:")
+        print(f"{host}/#/runs/details/{run_result.run_id}")
+        verify_func(run_id=run_result.run_id, run=run)
+
+    import fire
+    fire.Fire(main)
```
Comment (Ark-kun) [inline v2/test/samples/util.py:55]: Can you help me understand how does Fire help us here?

Comment (Bobgy) [reply v2/test/samples/util.py:55]: Sure!
Fire is used to provide CLI interface to run this sample, see https://github.com/kubeflow/pipelines/blob/1cd189567d4ba7e71aa1f31ddaf9a46c17ae8ae3/v2/test/components/run_sample.yaml#L35-L38.

**Comment by Ark-kun** [comment]
Is it possible to indicate the v2 samples in the name?

**Comment by Ark-kun** [comment]
Otherwise probably
/lgtm

**Comment by Bobgy** [comment]
> Is it possible to indicate the v2 samples in the name?

Good point! I think I will only name v2 only samples with v2 prefix

**Comment by Bobgy** [comment]
/approve
In fact, the two existing samples are compatible between v1 and v2, so we do not need a special mark.
We will need to think about how v2-only and v1-only samples should be organized, but I'll defer that to future considerations.

**Pull request merged**